### PR TITLE
Prebuild S2N

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*
- Prebuild S2N and update submodule for using s2n_cleanup_thread()

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .builder/actions/crt_size_check.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def run(self, env):
         # Maximum package size (for current platform) in bytes
         # NOTE: if you increase this, you might also need to increase the
         # limit in continuous-delivery/pack.sh
-        max_size = 7_500_000
+        max_size = 8_000_000
```
Comment (waahm7) [inline .builder/actions/crt_size_check.py:14]: Do we need to increase the limit in pack.sh?

Comment (TingDaoK) [reply .builder/actions/crt_size_check.py:14]: don't know... the pack.sh is not exactly match the CI here, you could try to run the test pipeline see if we need to

Comment (waahm7) [reply .builder/actions/crt_size_check.py:14]: We will find out

**Review by waahm7 — APPROVED** [review]

File: .builder/actions/crt_size_check.py (hunk 1 above)
Comment (waahm7) [reply .builder/actions/crt_size_check.py:14]: We found out in #587

**Pull request merged**